packages/ariakit-utils/src/__tests__/hooks-test.tsx

```diff
@@ -0,0 +1,407 @@
+    const result = useEvent(callback);
+    useEffect(() => {
+      effectFunction();
+    }, [result]);
+
+    return <button onClick={() => setCallback(jest.fn())} />;
+  };
+  const { container } = render(<TestComponent />);
+  expect(effectFunction).toBeCalledTimes(1);
+  fireEvent.click(container.querySelector("button")!);
+  expect(effectFunction).toBeCalledTimes(1);
+
+  // Check that error is thrown during render
+  let error: Error | undefined;
+  const TestComponent2 = () => {
+    const result = useEvent(jest.fn());
+
+    try {
+      result();
+    } catch (e) {
+      if (e instanceof Error) {
+        error = e;
+      }
+    }
+
+    return <div />;
+  };
+  expect(error).toBeUndefined();
+  render(<TestComponent2 />);
+  expect(error).toBeDefined();
+});
+
+test("useForkRef", () => {
+  const ref1 = jest.fn();
+  const ref2 = jest.fn();
+  const TestComponent = () => {
+    const internalRef = useRef();
+    const ref = useForkRef(internalRef, ref1, ref2);
+
+    return <button ref={ref} />;
+  };
+
+  const { container } = render(<TestComponent />);
+  expect(ref1).toBeCalledTimes(1);
+  expect(ref2).toBeCalledTimes(1);
+  const button = container.querySelector("button")!;
+  expect(ref1).toHaveBeenCalledWith(button);
+  expect(ref2).toHaveBeenCalledWith(button);
+
+  // No refs case
+  let refVal: any;
+  const TestComponent2 = () => {
+    const ref = useForkRef();
+    refVal = ref;
+    return <div />;
+  };
+  expect(refVal).toBeUndefined();
+  render(<TestComponent2 />);
+  expect(refVal).toBeUndefined();
+});
+
+test("useRefId", () => {
+  let idVal: string | undefined;
+  const TestComponent = () => {
+    const ref = useRef<HTMLDivElement>(null);
+    const id = useRefId(ref);
+    idVal = id;
+
+    return <div id="some-id" ref={ref} />;
+  };
+  expect(idVal).toBeUndefined();
+  render(<TestComponent />);
+  expect(idVal).toBe("some-id");
+
+  // Undefined case
+  let idVal2: string | undefined;
+  const TestComponent2 = () => {
+    const id = useRefId();
+    idVal = id;
+    return <div />;
+  };
+  expect(idVal2).toBeUndefined();
+  render(<TestComponent2 />);
+  expect(idVal2).toBeUndefined();
+});
+
+test("useId", () => {
+  // Case where react id is defined
+  const TestComponent = () => {
+    const [idState, setIdState] = useState("");
+    const id = useId(idState);
+
+    return <button onClick={() => setIdState("some-id")} id={id} />;
+  };
+  const { container } = render(<TestComponent />);
+  const button = container.querySelector("button")!;
+  expect(button.id).toBeTruthy();
+  fireEvent.click(button);
+  expect(button.id).toBe("some-id");
+});
+
+test("useTagName", () => {
+  let tagNameVal: string | undefined;
+  const TestComponent = () => {
+    const ref = useRef<HTMLDivElement>(null);
+    const tagName = useTagName(ref);
+    tagNameVal = tagName;
+
+    return <div ref={ref} />;
+  };
+  render(<TestComponent />);
+  expect(tagNameVal).toBe("div");
+
+  // With type case
```

Comment:
When a function has more than one test case, we can put it in another `test()`, for example:
```jsx
test("useTagName");
test("useTagName with default type");
```